frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/cloud-init-tab/cloud-init-tab.tsx

```diff
@@ -210,6 +211,13 @@ const CloudInitTabComponent: React.FC<ResultTabComponentProps> = ({
     }
     cloudInitData.set(key, value);
 
+    if (key === CloudInitDataFormKeys.HOSTNAME) {
+      onSettingsChange(
```

Comment:
on a second look. I would prefer to move this logic to `kubevirtInterOP` right before the request. It was done like that before and we can also parse the hostname from a custom script with much more ease.